File: git/baekjoon/미완/13172.cpp
#include <iostream>
long mod = 1000000007;
long pow(long a, long b)
{
  long r = 1;
  while(b)
  {
    if(b%2)
    {
      r = (r*a)%mod;
    }
    a = (a*a)%mod;
    b >>= 1;
  }
  return r;
}

int main()
{
  long m, res = 0, n, s;
  scanf("%ld", &m);
  while(m--) {
    scanf("%ld %ld", &n, &s);
    res += (s * pow(n, mod-2)) % mod;
  }
  printf("%ld\n", res % mod);
}

File: git/baekjoon/미완/13172.py
def jun_pow(base, exp, mod):
    res = 1
    while exp:
        if exp % 2:
            res = (res*base) % mod
        base = (base*base) % mod
        exp >>= 1
    return res

import sys
input = sys.stdin.readline
mod = 10**9+7
res = 0
for i in range(int(input())):
    n, s = map(int, input().split())
    res += (s*jun_pow(n, -1, mod) % mod)
print(res % mod)